### ISSUE TITLE
Differentiate between user-pressed and InputHelper-generated keystrokes?

### ISSUE THREAD

**Issue by ryderhaggard** [Visual-Vincent/InputHelper#6]
Is there a way for the hook to tell if its a user pressed key versus a input helper pressed  key?

sorry not sure how to flag this as a question instead of an issue.

**Comment by Visual-Vincent** [comment]
As a matter of fact there is! (sort of) Which was something I had overlooked in my previous implementation of the hooks.

As it turns out there is a way to check whether a keystroke was _injected by an application_ rather than by an input device, but unfortunately there is currently no way of knowing whether it came _specifically_ from InputHelper (e.g. it could come from a completely different application).

I just released v2.1.0.0 ([diff](https://github.com/Visual-Vincent/InputHelper/compare/v2.0.0.0...v2.1.0.0)) which adds the functionality to check for mouse/keyboard event injection. Go grab it over at [Releases](https://github.com/Visual-Vincent/InputHelper/releases).

Example usage:

**VB.NET**
```vb.net
Private Sub KeyboardHook_KeyDown(sender As Object, e As InputHelper.EventArgs.KeyboardHookEventArgs)
    'If the keystroke was not generated by the user, ignore it
    If e.Injected Then
        Return
    End If

    'Do something here...
End Sub
```

**C#**
```csharp
private void KeyboardHook_KeyDown(object sender, InputHelper.EventArgs.KeyboardHookEventArgs)
{
    // If the keystroke was not generated by the user, ignore it
    if(e.Injected)
        return;

    // Do something here...
}
```

As for flagging your issue as a question, I think that's only something maintainers and collaborators of the repository are able to do. Don't worry about it, I take care of it for every issue. :)
